### DISCUSSION TITLE
Access a specific cached type by id?

### DISCUSSION THREAD

**Discussion by corstian** [urql-graphql/urql#1489] [Q&A] [answered]
I'm in a position where I want to retrieve a specific (cached) object type by it's id. Assuming the type is available in the cache I suspect this must be possible, though have no idea how.

I have been playing around with this; but that doesn't seem fruitful;

```js
const client = useClient()

const q = gql`fragment _ on SomeType {
  id
  field1
  field2
}`;

console.log(q)

const result = client.readQuery(
  q,
  {
    id: "..."
  }
)

console.log(result)
```

Any ideas? Is this possible at all?

**Comment by JoviDeCroock** [comment, marked as answer]
I think you're looking for https://formidable.com/open-source/urql/docs/graphcache/local-resolvers/#reading-a-fragment

**Reply by corstian** [reply]
Close, but to my understanding the `cache` is not available outside of the configuration of the exchange. Thus to no use in ordinary components.

**Reply by kitten** [reply]
One of the rules of deterministic caching that we introduced (which is rather opinionated) is that you cannot access the cache outside of any usual operation (like a query or mutation) and furthermore not update it. So outside of the client you can never access it directly.

This has several benefits, but mainly it would help if you could describe us what you're trying to do. To me it sounds like you'd want your schema to allow you to access an item by ID, e.g. `Query.item(id: $id)`, since you're trying to "shim" your way around this, especially since if the cache doesn't have your item I'd assume your frontend would go into an unrecoverable state.

**Reply by corstian** [reply]
That clears things up! It's indeed some architectural thing which I'd like to cut short. I'm dealing with an `aircraft` query which retrieves a collection. The thing therewith is that `aircraft` is both singular as well as plural, and hoped that cache access would prevent myself from having to think about a naming pattern to separate one from another.

Maybe I'll just name the collection `aircrafts`, even though gramatically incorrect 😅
